File: Sources/custom_adapter/TopOnAdapter/Objc/6.5.x/Splash/AlxTopOnSplashEvent.m
//
//  AlxTopOnSplashEvent.m
//  AlxAdsOCDemo
//

#import "AlxTopOnSplashEvent.h"
#import "AlxTopOnBaseManager.h"
#import "AlxTopOnBiddingRequestManager.h"

static NSString *const TAG = @"AlxTopOnSplashEvent";

@implementation AlxTopOnSplashEvent

- (instancetype)initWithInfo:(NSDictionary *)serverInfo localInfo:(NSDictionary *)localInfo {
    self = [super initWithInfo:serverInfo localInfo:localInfo];
    return self;
}

- (void)splashAdDidLoad:(AlxSplashAd *)ad {
    NSLog(@"%@: splashAdDidLoad", TAG);
    if (self.isC2SBiding) {
        NSString *unitId = self.serverInfo[[AlxTopOnBaseManager unitID]];
        [AlxTopOnBiddingRequestManager disposeLoadSuccessWithPrice:[ad getPrice] unitID:unitId];
        self.isC2SBiding = NO;
    } else {
        NSMutableDictionary *adExtra = [NSMutableDictionary dictionary];
        adExtra[kATAdAssetsCustomObjectKey] = ad;
        [self trackSplashAdLoaded:ad adExtra:adExtra];
    }
}

- (void)splashAdDidFailToLoad:(AlxSplashAd *)ad didFailWithError:(NSError *)error {
    NSLog(@"%@: splashAdDidFailToLoad: %@", TAG, error.localizedDescription);
    if (self.isC2SBiding) {
        NSString *unitId = self.serverInfo[[AlxTopOnBaseManager unitID]];
        [AlxTopOnBiddingRequestManager disposeLoadFailWithError:error unitID:unitId];
    } else {
        [self trackSplashAdLoadFailed:error];
    }
}

- (void)splashAdDidShow:(AlxSplashAd *)ad {
    NSLog(@"%@: splashAdDidShow", TAG);
    [self trackSplashAdShow];
}

- (void)splashAdDidClick:(AlxSplashAd *)ad {
    NSLog(@"%@: splashAdDidClick", TAG);
    [self trackSplashAdClick];
}

- (void)splashAdDidClose:(AlxSplashAd *)ad {
    NSLog(@"%@: splashAdDidClose", TAG);
    [self trackSplashAdClosed:nil];
}

- (void)splashAdRenderDidFail:(AlxSplashAd *)ad didFailWithError:(NSError *)error {
    NSLog(@"%@: splashAdRenderDidFail: %@", TAG, error.localizedDescription);
    [self trackSplashAdShowFailed:error];
}

- (void)splashAdCountdown:(AlxSplashAd *)ad countdown:(NSInteger)countdown {
    NSLog(@"%@: splashAdCountdown: %ld", TAG, (long)countdown);
    [self trackSplashAdCountdownTime:countdown];
}

- (NSString *)networkUnitId {
    return self.serverInfo[[AlxTopOnBaseManager unitID]] ?: @"";
}

- (void)dealloc {
    NSLog(@"%@: dealloc", TAG);
}

@end

File: Sources/custom_adapter/TopOnAdapter/Swift/6.5.x/Splash/AlxTopOnSplashEvent.swift
//
//  AlxTopOnSplashEvent.swift
//  AlxAdsDemo
//

import Foundation
import AnyThinkSDK
import AlxAds

@objc(AlxTopOnSplashEvent)
public class AlxTopOnSplashEvent: ATSplashCustomEvent, AlxSplashAdDelegate {
    
    private static let TAG = "AlxTopOnSplashEvent"
    
    public override init(info serverInfo: [AnyHashable: Any], localInfo: [AnyHashable: Any]) {
        super.init(info: serverInfo, localInfo: localInfo)
    }
    
    public func splashAdDidLoad(_ ad: AlxSplashAd) {
        NSLog("%@: splashAdDidLoad", AlxTopOnSplashEvent.TAG)
        if self.isC2SBiding {
            AlxTopOnBiddingRequestManager.disposeLoadSuccess(price: ad.getPrice(), unitID: self.serverInfo[AlxTopOnBaseManager.unitID] as? String)
            self.isC2SBiding = false
        } else {
            var adExtra: [AnyHashable: Any] = [:]
            adExtra[kATAdAssetsCustomObjectKey] = ad
            self.trackSplashAdLoaded(adExtra)
        }
    }
    
    public func splashAdDidFailToLoad(_ ad: AlxSplashAd, didFailWithError error: Error) {
        NSLog("%@: splashAdDidFailToLoad: %@", AlxTopOnSplashEvent.TAG, error.localizedDescription)
        if self.isC2SBiding {
            AlxTopOnBiddingRequestManager.disposeLoadFail(error: error, unitID: self.serverInfo[AlxTopOnBaseManager.unitID] as? String)
        } else {
            self.trackSplashAdLoadFailed(error)
        }
    }
    
    public func splashAdDidShow(_ ad: AlxSplashAd) {
        NSLog("%@: splashAdDidShow", AlxTopOnSplashEvent.TAG)
        self.trackSplashAdShow()
    }
    
    public func splashAdDidClick(_ ad: AlxSplashAd) {
        NSLog("%@: splashAdDidClick", AlxTopOnSplashEvent.TAG)
        self.trackSplashAdClick()
    }
    
    public func splashAdDidClose(_ ad: AlxSplashAd) {
        NSLog("%@: splashAdDidClose", AlxTopOnSplashEvent.TAG)
        self.trackSplashAdClosed(nil)
    }
    
    public func splashAdRenderDidFail(_ ad: AlxSplashAd, didFailWithError error: Error) {
        NSLog("%@: splashAdRenderDidFail: %@", AlxTopOnSplashEvent.TAG, error.localizedDescription)
    }
    
    public func splashAdCountdown(_ ad: AlxSplashAd, countdown: Int) {
        NSLog("%@: splashAdCountdown: %d", AlxTopOnSplashEvent.TAG, countdown)
        self.trackSplashAdCountdownTime(countdown)
    }
    
    public override var networkUnitId: String {
        get {
            return self.serverInfo[AlxTopOnBaseManager.unitID] as? String ?? ""
        }
        set {
        }
    }
    
    deinit {
        NSLog("%@: deinit", AlxTopOnSplashEvent.TAG)
    }
}
